Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts

```diff
@@ -8,26 +8,27 @@ import { Observable, of } from 'rxjs';
   providedIn: 'root'
 })
 export class CurrentUserService {
-  private permissions: Array<string> = [];
-  private lastUpdateTimestamp = 0;
+  private permissionMap: Map<string, Array<string>> = new Map<string, Array<string>>();
+  private lastUpdateTimestampMap: Map<string, number> = new Map<string, number>();
 
   constructor(private authUserService: FabricAuthUserService) {
   }
 
-  getPermissions(): Observable<string[]> {
+  getPermissions(securableItem: string): Observable<string[]> {
     const currentTimeStamp = new Date().getTime();
-    if (this.lastUpdateTimestamp === 0 || (Math.abs(currentTimeStamp - this.lastUpdateTimestamp) / 60000) > 2) {
-      this.lastUpdateTimestamp = currentTimeStamp;
-      return this.authUserService.getCurrentUserPermissions().pipe(tap(userPermissionResponse => {
-        this.permissions = userPermissionResponse.permissions;
+    if (!this.lastUpdateTimestampMap.has(securableItem)
```

Comment:
This is starting to get complex.  Maybe throwing this into a new function would be good.  Perhaps this logic is an object?

Just a thought

Reply:
Good call - I just incorporated that suggestion.